libs/ui/src/lib/TextField.tsx

```diff
@@ -0,0 +1,136 @@
+import styled from '@emotion/styled';
+import React, {
+  forwardRef,
+  InputHTMLAttributes,
+  ReactNode,
+  Ref,
+  useState,
+} from 'react';
+import { ThemeProps } from '../color';
+import { useBoolean } from '../hooks/useBoolean';
+
+export type ValidationState = 'valid' | 'invalid';
+
+export interface Validation {
+  validationState?: ValidationState;
+  isRequired?: boolean;
+}
+
+interface Props extends InputHTMLAttributes<HTMLInputElement>, Validation {
+  label?: ReactNode;
+  type?: string;
+}
+
+const ControlledTextField = forwardRef(
+  ({ type = 'text', label, ...rest }: Props, ref: Ref<HTMLInputElement>) => {
```

Comment:
we extract `type` in `Props` type

this component is `****TextField`, right?

```ts

```
```suggestion
interface TemporaryProps extends InputHTMLAttributes<HTMLInputElement>, Validation {
  label?: ReactNode;
}

type Props = Omit<TemporaryProps, 'type'>;

const ControlledTextField = forwardRef(
  ({ label, ...rest }: Props, ref: Ref<HTMLInputElement>) => {
```